### ISSUE TITLE
Doubt: relay change event data without compromising the signature

### ISSUE THREAD

**Issue by frbitten** [nostr-protocol/nips#485]
Are there any fields or information in the event that the relay can change without invalidating the signature?

For example any event metadata?
Some situations that I find interesting for the relay to change auxiliary data of the event:

1. The original event has no indication of the preferred relay and when propagating the relay it could be added to the event and thus facilitate the operation of clients with more complete events.
2. Update to newer functionality.
3. I have an event that has numerous boosted, currently the client needs to receive the event and all its boosted events to count and show on the internet. The relay could already do this accounting internally and send the main note to the client already with the information on the number of boosted made.

Is there already something like this or some way to include this dynamic information in the event to facilitate the client side?

**Comment by fiatjaf** [comment]
Yes, the relay (or anyone) can add any fields it want to the event object without compromising the signature. The signature only covers the fields specified in NIP-01.

**Comment by frbitten** [comment]
OK. But the tag region is defined in NIP-01, so all tags are used to create the signature. As a result, the relay cannot include any tags on its own. Correct?

Wouldn't it be interesting to have an area in the event similar to the tags for use by relays that doesn't affect the signature?

**Comment by fiatjaf** [comment]
Correct.

Yes, I think that would be interesting.

**Comment by fiatjaf** [comment]
Do we need a NIP on that? I think we need a use case first, and then we make a NIP for that use case, like NIP 3.
